### ISSUE TITLE
Polygon extrusion heights stretch as map moves

### ISSUE THREAD

**Issue by rbrundritt** [mapbox/mapbox-gl-js#8821]
[Extruded Polygon Issue.zip](https://github.com/mapbox/mapbox-gl-js/files/3666475/Extruded.Polygon.Issue.zip)
I noticed recently that when zoomed out with extruded polygons, if I pan the map up and down, the height of the extrusions changes. If you rotate the map 90 degrees and pan left/right, the same effect occurs. 

**mapbox-gl-js version**:  1.4.0

**browser**: Chrome

### Steps to Trigger Behavior

 1. Sample attached
 2. download and unzip code and add key to index.html [Extruded Polygon Issue.zip](https://github.com/mapbox/mapbox-gl-js/files/3666480/Extruded.Polygon.Issue.zip)
 3. npm install
 4. npm start
 5. Pan map north/south and watch the height of the polygons.

(sample too big for jsbin so attached zip)

### Expected Behavior

As you move the map, extruded polygons should have a consistent height. 

### Actual Behavior

They appear to be scaling based on the latitude. Here is an animation of what happens.

![ExtrudedPolygonIssue](https://user-images.githubusercontent.com/6459716/65828814-f3aee880-e253-11e9-8d94-2df7a1c9e2b6.gif)

I suspect the height measurement is based on the meters per pixels of the center of the map. For a more consistent experience it might be better to use the meters per pixel value for 0 degrees latitude. If backwards compatibility is needed (perhaps this effect is desired for buildings when zoomed in close) maybe introducing a style options would make sense.

**Comment by ansis** [comment]
>  I suspect the height measurement is based on the meters per pixels of the center of the map.

Yep, this is what is happening. And you're right that this was designed for buildings when zoomed in close.

I experimented with disabling this behavior at low zoom levels and easing into it at higher zoom levels but I think this will introduce a weird transition period which we probably don't want. https://github.com/mapbox/mapbox-gl-js/compare/fill-extrusion-scaling-experiment

> maybe introducing a style options would make sense.

Would being able to specify the height in pixels work for you?

Another thing that would fix this would support for a projection that doesn't distort scale.

**Comment by rbrundritt** [comment]
Looking into this more, a pixel height option may be good for some scenarios where the user doesn't mind the heights changing relative to the zoom level. Other may want to keep the heights a constant height regardless of zoom level. If an option is exposed using either of these methods, one performance improvement is that the render heights wouldn't need to be recalculated on pan, only zoom.

**Comment by vallendm** [comment]
We've also noticed polygon extrusion heights altering when the map is panned north south. I just tried using a build based off https://github.com/mapbox/mapbox-gl-js/compare/fill-extrusion-scaling-experiment - and the results looked better and more consistent with our use case ( which is either a either extruding geographic feature or hexbin regions - where there accuracy of the height of the region is probably secondary to relative heights between regions). Having an option so that heights are constant would definately be something we would be interested in.

**Comment by dantish** [comment]
Hi guys!
Wanted to kindly remind -- are there any plans for this feature?
Thanks in advance!
